File: medium/7. Reverse integer/Main.java
void main() {
    IO.println(reverse(123));
}

public int reverse(int x) {
    int result = 0;
    while (x != 0) {
        if (((long) result) * 10 > Integer.MAX_VALUE) {
            return 0;
        }
        if (((long) result) * 10 < Integer.MIN_VALUE) {
            return 0;
        }
        result *= 10;
        result += x % 10;
        x /= 10;
    }

    return result;
}

File: medium/7. Reverse integer/main.go
package main

import (
	"fmt"
	"math"
)

func reverse(x int) int {
	result := 0
	for x != 0 {
		pop := x % 10
		x /= 10

		if int64(result)*10 > math.MaxInt32 {
			return 0
		}
		if int64(result)*10 < math.MinInt32 {
			return 0
		}

		result = result*10 + pop
	}
	return result
}

func main() {
	fmt.Println(reverse(123))
}
